backend/packages/Upgrade/src/api/middlewares/UserCheckMiddleware.ts

```diff
@@ -0,0 +1,36 @@
+import * as express from 'express';
+import { SettingService } from '../services/SettingService';
+import { SERVER_ERROR } from 'upgrade_types';
+import { AppRequest } from '../../types';
+import { Service } from 'typedi';
+import { ExperimentUserService } from '../services/ExperimentUserService';
+
+@Service()
+export class UserCheckMiddleware {
+  constructor(public settingService: SettingService, public experimentUserService: ExperimentUserService) {}
+
+  public async use(req: AppRequest, res: AppRequest, next: express.NextFunction): Promise<any> {
+    try {
+      const user_id = req.get('User-Id');
+      req.logger.child({ user_id });
+      req.logger.debug({ message: 'User Id is:', user_id });
+      console.log(user_id);
+      if (req.url !== '/api/init') {
```

Comment:
There is no need to check userId from the header for the init call, so bring this if statement on the top.